### DISCUSSION TITLE
Equivalent of typeorm's lazy relation?

### DISCUSSION THREAD

**Discussion by ldiqual** [prisma/prisma#5951] [Q&A] [answered]
What would be / is there a prisma equivalent to TypeORM's lazy relation loading?

```ts
const user = await User.findOneOrFail(1)
const company = await user.company
```

I am not asking about loading the company when fetching the user, but rather fetching the company later down the call-chain.

One option would be:

```ts
const user = await prisma.users.findUnique({ where: { id: 1 } })
const company = await prisma.users.findUnique({ where: { id: user.id } }).company()
```

But this would be pretty verbose and repetitive if I need to do this in many helpers.

**Comment by ryands17** [comment, marked as answer]
@ldiqual 👋 

Prisma doesn't have a class based architecture like TypeORM so currently there is no way of fetching the `user` first and then lazily fetching the `company`. Whenever you fetch a value from Prisma, it returns a plain JS object and not some custom object of a class with methods.

Currently the only way to do this would be something like the one above. I would suggest creating a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) so that we can look into this :)

**Reply by ldiqual** [reply]
Thank you! Filed here https://github.com/prisma/prisma/issues/5962

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
